app/javascript/packs/colorPreview.js

```diff
@@ -0,0 +1,57 @@
+const colorField = {
+  // input type="color"
+  bgColor: document.getElementById('bg-color-colorfield'),
+  textColor: document.getElementById('text-color-colorfield'),
+};
+
+const textField = {
+  // input type="text"
+  bgColor: document.getElementById('bg-color-textfield'),
+  textColor: document.getElementById('text-color-textfield'),
+};
+
+const preview = document.getElementById('color-select-preview-logo');
+
+// Assigns input[type='text'] values to input[type='color']
+function updateColorFields() {
+  colorField.bgColor.value = textField.bgColor.value;
+  colorField.textColor.value = textField.textColor.value;
+}
+
+// Updating text fields when color fields are changed
+function updateTextFields() {
+  textField.bgColor.value = colorField.bgColor.value;
+  textField.textColor.value = colorField.textColor.value;
+}
+
+// Updates Preview Colors
+function updatePreview() {
+  preview.style.backgroundColor = textField.bgColor.value;
+  preview.style.fill = textField.textColor.value;
+}
+
+// Event Watchers
+// When color fields change -> updateTextField values and refresh preview
+function watchColorFields() {
+  updateTextFields();
+  updatePreview();
+}
+
+// When text fields change -> updateColorField values and refresh preview
+function watchTextFields(e) {
+  if (e.target.value.match(/#[0-9a-f]{6}/gi)) {
+    updateColorFields();
+    updatePreview();
+  }
+}
+
+// Event Listeners
+colorField.bgColor.addEventListener('input', watchColorFields);
+colorField.textColor.addEventListener('input', watchColorFields);
+textField.bgColor.addEventListener('keyup', watchTextFields);
+textField.textColor.addEventListener('keyup', watchTextFields);
+
+// on init
+updateColorFields();
+updatePreview();
+preview.style.display = 'inline-block';
```

Comment:
This should probably be inside the `window.InstantClick.on('change', () => {` block, see the colorPicker.js

Reply:
Hi I added it inside InstantClick change event. Also, just adding it inside InstantClick block didn't work so had to call it from outside as well. Even in `colorPicker.js` there is similar code outside the block and inside the change block. 

***colorPicker.js***
```js
const colorInputs = document.getElementsByClassName('color-picker');
const colorPalettes = document.getElementsByClassName('color-palette');

for (let i = 0; i < colorInputs.length; i += 1) {
  makeColorPickerGo(colorInputs[i], colorPalettes[i]);
}

window.InstantClick.on('change', () => {
  const colorInputElements = document.getElementsByClassName('color-picker');
  const colorPaletteElements = document.getElementsByClassName('color-palette');

  for (let i = 0; i < colorInputElements.length; i += 1) {
    makeColorPickerGo(colorInputElements[i], colorPaletteElements[i]);
  }
});

```

Just wondering when exactly InstantClick is fired? It is used in a lot of files so hard to figure out